### PR TITLE
DRAFT: 96 tugboat acquia

### PR DESCRIPTION
**Pull request by edutrul**
Add a `tugboat.acquia-example.yml` file that includes memcache.

- [ ] Go to composer.json and place:
```
"extra": {
     "drainpipe": {
         "tugboat": ["acquia"]
     }
}
```

- [ ]  run the command

What it does is it creates a file `tugboat.acquia-example.yml`, which contains the memcached.

### CODE REVIEW

File: scaffold/tugboat/tugboat.acquia-example.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      # Commands that import files, databases,  or other assets. When an
+      # existing preview is refreshed, the build workflow starts here,
+      # skipping the init step, because the results of that step will
+      # already be present.
+      update:
+        - echo "Nothing to do as we import databases from the Drupal service."
+
+  memcached:
+
+    image: tugboatqa/memcached:1
```
Comment (deviantintegral) [inline scaffold/tugboat/tugboat.acquia-example.yml:106]: Wow, we literally only need two lines here? I figured there would be a tiny bit more.

Seeing this, part of me thinks we could instead combine multiple files together, so we don't have to duplicate the first 100 lines. However, I'm not sure that we won't need _something_ custom for environment variables to do something like downsync Acquia databases into base previews.

File: src/ScaffoldInstallerPlugin.php [hunk 2]
```diff
@@ -273,6 +273,15 @@ private function installCICommands(): void
                     }
                 }
             }
+            foreach ($this->extra['drainpipe']['tugboat'] as $tugboat) {
+                if ($tugboat === 'acquia') {
```
Comment (deviantintegral) [inline src/ScaffoldInstallerPlugin.php:277]: What do you think about renaming `$tugboat` to `$provider`?

Comment (edutrul) [reply src/ScaffoldInstallerPlugin.php:277]: Yeah you are right at the end they are providers and we can have as many other providers. Perfect makes sense thank you, I will do it.

File: scaffold/tugboat/tugboat.acquia-example.yml (hunk 1 above)
Comment (edutrul) [reply scaffold/tugboat/tugboat.acquia-example.yml:106]: _Seeing this, part of me thinks we could instead combine multiple files together, so we don't have to duplicate the first 100 lines. However, I'm not sure that we won't need something custom for environment variables to do something like downsync Acquia databases into base previews._

I don't understand your comment, could you please explain a little more. Thank you

**Comment by q0rban** [comment]
This might be helpful: https://github.com/TugboatQA/docs/issues/207

I would recommend adding the parts to increase the innodb log size and max allowed packet.

**Comment by edutrul** [comment]
Thank you @q0rban I'll add it when no parameter is added like tugboat or acquia. Awesome plan. Thanks again.

**Pull request closed without merging**